src/lumigo_tracer/spans_container.py

```diff
@@ -150,13 +150,12 @@ def add_unparsed_request(self, parse_params: HttpRequest):
         """
         if self.http_spans:
             last_event = self.http_spans[-1]
-            if last_event and last_event.get("type") == HTTP_TYPE:
+            if last_event and last_event.get("type") == HTTP_TYPE and self.previous_request:
                 if last_event.get("info", {}).get("httpInfo", {}).get("host") == parse_params.host:
                     if "response" not in last_event["info"]["httpInfo"]:
                         self.http_spans.pop()
-                        prev_headers, prev_body = self.previous_request
-                        body = (prev_body + parse_params.body)[:MAX_BODY_SIZE]
-                        self.add_request_event(parse_params.clone(headers=prev_headers, body=body))
```

Comment:
we didn't clone the `uri` from the previous request.
This is generally smells - we need to clone the previous request (that has all the data) and just override the new body.
That's what I did :)